### ISSUE TITLE
 Check failed: proto.SerializeToOstream(&output)

### ISSUE THREAD

**Issue by zhangqiudan** [CLT29/OpenSALICON#2]
Hi, everyone.
I used salicon-net (finetune-salicon.py)to  try to train a new model over my own dataset.
However, There is a error when I train my model.
error:
I1122 09:19:42.497810 16306 net.cpp:761] Ignoring source layer sec_pool5
I1122 09:19:42.497845 16306 net.cpp:761] Ignoring source layer custom_interpolation_layer
working on 0 of 600
I1122 09:19:42.906407 16306 solver.cpp:228] Iteration 0, loss = 3359.02
I1122 09:19:42.906457 16306 solver.cpp:244]     Train net output #0: loss = 3359.02 (* 1 = 3359.02 loss)
I1122 09:19:42.906472 16306 sgd_solver.cpp:106] Iteration 0, lr = 1e-06
F1122 09:19:43.967656 16306 io.cpp:69] Check failed: proto.SerializeToOstream(&output) 
*** Check failure stack trace: ***
Aborted (core dumped)
I'm looking forward to your reply. Thank you so much

**Comment by zhangqiudan** [comment]
@CLT29  When I use cpu model to train, it success. But it down when I use gpu model.

**Comment by CLT29** [comment]
This seems to be an issue with caffe. Did you check the solver prototxt and make the snapshot directory?

https://github.com/BVLC/caffe/issues/1394

-chris

-------- Original message --------
From: zhangqiudan
Date:11/21/2016 8:30 PM (GMT-05:00)
To: CLT29/OpenSALICON
Subject: [CLT29/OpenSALICON] Check failed: proto.SerializeToOstream(&output) (#2)

Hi, everyone.
I used salicon-net (finetune-salicon.py)to try to train a new model over my own dataset.
However, There is a error when I train my model.
error:
I1122 09:19:42.497810 16306 net.cpp:761] Ignoring source layer sec_pool5
I1122 09:19:42.497845 16306 net.cpp:761] Ignoring source layer custom_interpolation_layer
working on 0 of 600
I1122 09:19:42.906407 16306 solver.cpp:228] Iteration 0, loss = 3359.02
I1122 09:19:42.906457 16306 solver.cpp:244] Train net output #0: loss = 3359.02 (\* 1 = 3359.02 loss)
I1122 09:19:42.906472 16306 sgd_solver.cpp:106] Iteration 0, lr = 1e-06
F1122 09:19:43.967656 16306 io.cpp:69] Check failed: proto.SerializeToOstream(&output)
**\* Check failure stack trace: ***
Aborted (core dumped)
I'm looking forward to your reply. Thank you so much

##

**Comment by zhangqiudan** [comment]
I have checked  the solver prototxt and made the snapshot directory , but it still occurred this error. And then I used the original prototxt downloaded from the Github and set cpu pattern to run, it success.

**Comment by zhangqiudan** [comment]
@CLT29

**Comment by zhangqiudan** [comment]
Do I need to change the original prototxt to savethe caffemodel ?

**Comment by CLT29** [comment]
Can you post your solver prototxt file?
-chris

**Comment by CLT29** [comment]
By the way, you have verified that you have enough disk space to save the snapshots (which can be 1GB each)?

-chris

**Comment by zhangqiudan** [comment]
yeah, I have verified the disk space is enough.  
Solver:
train_net: "finetune_salicon.prototxt"
base_lr: 0.000001
lr_policy: "step"
gamma: 0.1
stepsize: 4000
display: 20
#average_loss: 100
iter_size: 1
momentum: 0.9
weight_decay: 0.0005
# We disable standard caffe solver snapshotting and implement our own snapshot
# function
snapshot: 0
# We still use the snapshot prefix, though
snapshot_prefix: "finetuned_salicon"
#debug_info: true
solver_mode: GPU
solver_type: ADADELTA
delta: 1e-6
I did not change the solver prototxt.

**Comment by CLT29** [comment]
Can you confirm if this still happens if you remove the solver_type: ADADELTA and delta: 1e-6 lines?
-chris

**Comment by zhangqiudan** [comment]
I just tried to remove the solver_type,  It can run successfully. 
Thank you very much for solving the problem for me.
Best wishes to you.
by qiudan

**Comment by CLT29** [comment]
You may want to try adjusting the learning rate (maybe slightly increasing it), since now you will be using SGD instead of ADADELTA. Be careful to monitor the loss to make sure the loss doesn't diverge and decreases if you change the learning rate. You may want to also try a different solver_type, such as ADAM.

Good luck. You may want to report this problem to Caffe, I do not know what causes this for you.

-chris

-------- Original message --------
From: zhangqiudan
Date:11/22/2016 3:59 AM (GMT-05:00)
To: CLT29/OpenSALICON
Cc: Chris Thomas , Mention
Subject: Re: [CLT29/OpenSALICON] Check failed: proto.SerializeToOstream(&output) (#2)

I just tried to remove the solver_type, It can run successfully.
Thank you very much for solving the problem for me.
Best wishes to you.
by qiudan

**Comment by zhangqiudan** [comment]
ok, I will do that. Thanks for you advises. And I will try to learn a model over my own dataset.
Best wishes.

**Comment by zhangqiudan** [comment]
ok, I will do that. Thanks for your advises. And I will try to learn a model over my own dataset.
Best wishes.
